core/src/main/scala/org/apache/spark/ui/jobs/JobProgressListener.scala

```diff
@@ -73,8 +73,15 @@ class JobProgressListener(conf: SparkConf) extends SparkListener with Logging {
   var numCompletedStages = 0
   var numFailedStages = 0
 
+  // Executors:
+  val addedExecutors = TreeSet[(Long, ExecutorId)]()
```

Comment:
Rather than modifying the JobProgressListener, can you use the existing ExecutorsListener? Feel free to modify it if you need to.